### ISSUE TITLE
MoGA: Searching Beyond MobileNetV3

### ISSUE THREAD

**Issue by sile** [sile/readings#75]
URL
-----

https://arxiv.org/abs/1908.01314

Abstract
----------

> The evolution of MobileNets has laid a solid foundation for neural network applications on mobile end. With the latest MobileNetV3, neural architecture search again claimed its supremacy in network design. Unfortunately, till today all mobile methods mainly focus on CPU latencies instead of GPU, the latter, however, is much preferred in practice for it has faster speed, lower overhead and less interference. Bearing the target hardware in mind, we propose the first Mobile GPU-Aware (MoGA) neural architecture search in order to be precisely tailored for real-world applications. Further, the ultimate objective to devise a mobile network lies in achieving better performance by maximizing the utilization of bounded resources. Urging higher capability while restraining time consumption is not reconcilable. We alleviate the tension by weighted evolution techniques. Moreover, we encourage increasing the number of parameters for higher representational power. With 200x fewer GPU days than MnasNet, we obtain a series of models that outperform MobileNetV3 under the similar latency constraints, i.e., MoGA-A achieves 75.9% top-1 accuracy on ImageNet, MoGA-B meets 75.5% which costs only 0.5 ms more on mobile GPU. MoGA-C best attests GPU-awareness by reaching 75.3% and being slower on CPU but faster on GPU.The models and test code is made available here [this https URL](https://github.com/xiaomi-automl/MoGA).

Table of Contents
--------------------

- (1) Introduction
- (2) Related Works
- (3) Mobile GPU-Aware NAS Based on Multi-Objective Optimization
  - (3-1) Mobile GPU Awareness
  - (3-2) Underfitting and Overfitting
  - (3-3) Problem Formulation
- (4) Solving it Using Weighted NSGA-II
  - (4-1) Search Space
  - (4-2) Accuracy and Latency Prediction
  - (4-3) Weighted NSGA-II
  - (4-4) Our NAS Pipeline
- (5) Experiments
  - (5-1) Mobile GPU Latency
  - (5-2) Training
  - (5-3) Comparisons with State-of-the-art Methods
  - (5-4) Mobile GPU Awareness Analysis
  - (5-5) GPU Cost Analysis with More Mobile Devices
- (6) Results
  - (6-1) Ablation Study
    - Model Selection
    - Does it matter to use parameters as an objective?
- (7) Conclusion

BibTeX
------

```bibtex
@article{chu2019moga,
  title={MoGA: Searching Beyond MobileNetV3},
  author={Chu, Xiangxiang and Zhang, Bo and Xu, Ruijun},
  journal={arXiv preprint arXiv:1908.01314},
  year={2019}
}
```

See Also
----------

- GitHub: https://github.com/xiaomi-automl/MoGA
